src/ViewInstance.js

```diff
@@ -414,6 +411,49 @@ class ViewInstance extends React.Component {
     this.setState({ isImportRecordModalOpened: false });
   }
 
+  checkInstanceSharingProgress = ({ sourceTenantId, instanceIdentifier }) => {
+    return this.props.mutator.shareInstance.GET({
+      params: { sourceTenantId, instanceIdentifier },
+      headers: {
+        [OKAPI_TENANT_HEADER]: this.props.centralTenantId,
+        'Content-Type': 'application/json',
+        ...(this.props.okapi.token && { 'X-Okapi-Token': this.props.okapi.token }),
+      },
+    });
+  }
+
+  waitForInstanceSharingComplete = ({ sourceTenantId, instanceIdentifier, instanceTitle }) => {
+    return new Promise((resolve, reject) => {
+      const interval = setInterval(() => {
+        const onError = error => {
+          this.calloutRef.current.sendCallout({
+            type: 'error',
+            message: <FormattedMessage id="ui-inventory.shareLocalInstance.toast.unsuccessful" values={{ instanceTitle }} />,
+          });
+
+          reject(error);
+        };
+        const onSuccess = response => {
+          const sharingStatus = response?.sharingInstances[0]?.status;
+
+          if (sharingStatus === INSTANCE_SHARING_STATUSES.COMPLETE) {
+            clearInterval(interval);
+            resolve(response);
+          }
+
+          if (sharingStatus === INSTANCE_SHARING_STATUSES.ERROR) {
+            clearInterval(interval);
+            onError(response);
+          }
+        };
+
+        this.checkInstanceSharingProgress({ sourceTenantId, instanceIdentifier })
+          .then(onSuccess)
+          .catch(onError);
```

Comment:
`onError` is called in two places but `clearInterval` is invoked once, let's move `clearInterval` to `onError` cb.

Reply:
done